doc/source/how-to/packaging.rst

```diff
@@ -322,6 +322,51 @@ behavior that the user would expect:
    Install 'matplotlib' with:
    pip install matplotlib
 
+Optional target recommendations in the PyAnsys ecosystem
+--------------------------------------------------------
+
+PyAnsys projects, for harmonization purposes, should use the following targets or
+optional dependency groups:
```

Comment:
From my understanding, we are only talking about extras (https://packaging.python.org/en/latest/specifications/dependency-specifiers/#extras) here, not dependency groups (https://packaging.python.org/en/latest/specifications/dependency-groups/#dependency-groups).

We could _additionally_ clarify that development-only dependencies should _ideally_ be specified in dependency groups, or as extras if dependency groups aren't available with the build system used.

Reply:
That's a good point, yeah -- I'll add the clarification. We are only talking about extras here, but on build systems such as flit there is no distinction AFAIK between extras and dependency-groups.

Reply:
> there is no distinction AFAIK between extras and dependency-groups

There _has_ to be a distinction, since extras go into the `.whl`, whereas `dependency-groups` do not.

Looking at this https://flit.pypa.io/en/stable/pyproject_toml.html#dependencies, maybe only extras are supported?

I guess the naming is inconsistent, seems flit calls extras `optional-dependencies`.

Reply:
> There _has_ to be a distinction, since extras go into the `.whl`, whereas `dependency-groups` do not.

My point was that ``dependency-groups`` are not supported in flit, meaning that all of them are considered extras.

Reply:
Right, agreed - if `dependency-groups` is not supported falling back to extras is a decent enough solution.

The `dependency-groups` [PEP](https://peps.python.org/pep-0735/) is fairly recent (accepted Oct'24), so it's expected that support isn't universal.